test/index.spec.js

```diff
@@ -0,0 +1,165 @@
+/* eslint-disable max-nested-callbacks */
+
+'use strict';
+
+const expect = require('chai').expect;
+const sinon = require('sinon');
+
+const taf = require('../index');
+
+describe('timeoutify-async-function', () => {
+  let clock;
+  let sandbox;
+
+  const mockDeadlineTimeout = 5000;
+  const mockCompletionTimeout = 2500;
+  const mockOverrunTimeout = 7500;
+
+  let stubFn;
+
+  before(() => {
+    sandbox = sinon.sandbox.create();
+  });
+
+  beforeEach(() => {
+    clock = sinon.useFakeTimers();
+
+    stubFn = sandbox.stub();
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+    clock.restore();
+  });
+
+  describe('timeoutify', () => {
+    it('wraps a function in a timed-promise', () => {
+      const result = taf(stubFn, mockDeadlineTimeout);
+      expect(result).to.be.a('function');
+    });
+
+    it('resolves if the function resolves in the time allotted', (done) => {
+      (async () => {
```

Comment:
Open question (I don't know the answer): Given that mocha can handle returned promises without the use of `done`, can we do something here with `async` to similarly not need `done`?